napari/layers/_base_layer/model.py

```diff
@@ -237,6 +269,16 @@ def cursor_size(self, cursor_size):
         self.viewer.cursor_size = cursor_size
         self._cursor_size = cursor_size
 
+    @property
+    def scale_factor(self):
+        """float: Conversion factor from canvas coordinates to image
+        coordinates. Depends on the current zoom level.
+        """
+        transform = self._node.canvas.scene.node_transform(self._node)
+        scale_factor = transform.map([1, 1])[:2] - transform.map([0, 0])[:2]
+
+        return scale_factor.mean()
```

Comment:
I feel like this should be either the same in each dimension, in which case, rather than the mean, just pick one, or allowed to be different in each dimension, in which case, we should return both.

Reply:
Also, the fact that it depends on the current zoom level suggests one more set of coordinates: screen coordinates. Or does zoom mean level of detail here for multiscale pyramids?

Reply:
This is really screen coordinates, shouldn't have called it canvas. Zoom does not relate to the image pyramids directly (though you could use this value to determine which level of the pyramid to show). I've improved the doc string.

Also both values should be the same so I've just taken the first